extension/jaegerremotesampling/go.mod

```diff
@@ -22,11 +22,11 @@ require (
 	go.opentelemetry.io/collector/featuregate v1.22.1-0.20241220212031-7c2639723f67
 	go.uber.org/goleak v1.3.0
 	go.uber.org/zap v1.27.0
+	golang.org/x/net v0.33.0
```

Comment:
There shouldn't be changes like this in go mod, they are not related to the pr.

Reply:
`x/net/context` was being used in the grpc_handler_test file, which caused this change to `go.mod`.
Changed to use the stdlib `context`.